### PR TITLE
RATEST-143: Fix record merge issue test

### PR DESCRIPTION
**Pull request by kdaud**
Ticket[ ID](https://issues.openmrs.org/browse/RATEST-143)
Description -> Fix record merge issue test

### CODE REVIEW

File: ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java [hunk 1]
```diff
@@ -80,6 +83,5 @@ public void tearDown() throws Exception {
         p.uuid = patient.uuid;
         deletePatient(p);
         waitForPatientDeletion(patient.uuid);
-        headerPage.logOut();
```
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:83]: After deleting the test patient data from the `db instance` there is no need to invoke `headerPage.logOut()`  within `tearDown() ` function

File: ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java [hunk 2]
```diff
@@ -41,9 +43,10 @@ public void setUp() throws Exception {
         patient1 = new TestPatient();
     }
 
-    @Ignore
+    @Test
+    @Category(BuildTests.class)
     public void recordMergeIssueTest() throws Exception {
-        homePage.goToRegisterPatientApp();
+        homePage.goToRegisterPatientApp().waitForPage();
```
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:48]: After the test invoking many pages during the test `setUp()`, the patient registration  page takes a bit longer to be loaded  by the `classLoader`. Adding `waitforPage()` resolves the delay **Otherwise** the actions that follow are not effective and the patient data is not saved in `db instance`

File: ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java (hunk 1 above)
Comment (sherrif10) [reply ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:83]: Why not Adding header.logout, do you mean after deletion of patient, and the test wait to delete a patient  the testpasses without logging out ??

Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:83]: > Why not Adding header.logout

This being a `teardown` function, I think there is no need to invoke `headerPage.logout()`. Basing on the legacy ui test `workflow`, the test comes to an end when the `test patient data` has been deleted from `db instance` by the logic of `RestClient`. By the time   `headerPage.logOut()` is invoked, the` test patient data` has been already deleted from the `db instance` at  LN 84 & 85


> do you mean after deletion of patient, and the test wait to delete a patient the testpasses without logging out ??

After simulation on the test without invoking  `headerPage.logOut()` within the `teardown` function, the test `passes`.

File: ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java [hunk 3]
```diff
@@ -41,9 +42,10 @@ public void setUp() throws Exception {
         patient1 = new TestPatient();
     }
 
-    @Ignore
+    @Test
+    @Category(BuildTests.class)
     public void recordMergeIssueTest() throws Exception {
-        homePage.goToRegisterPatientApp();
```
Comment (sherrif10) [inline ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:46]: is this change necessary,

Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/RecordMergeIssueTest.java:46]: > is this change necessary,

Explanation about the change is already [here](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/461#discussion_r654835119)
